pkg/controller/drain/drain_controller.go

```diff
@@ -356,16 +402,45 @@ func (ctrl *Controller) drainNode(node *corev1.Node, drainer *drain.Helper) erro
+		err := upgrademonitor.GenerateAndApplyMachineConfigNodes(&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateExecuted, Reason: string(v1alpha1.MachineConfigNodeUpdateCordoned), Message: fmt.Sprintf("(Un)Cordoning Node as part of In progress upgrade phase")},
+			&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateCordoned, Reason: fmt.Sprintf("%s%s", string(v1alpha1.MachineConfigNodeUpdateExecuted), string(v1alpha1.MachineConfigNodeUpdateCordoned)), Message: fmt.Sprintf("(Un)Cordoning node. This process will not be complete untill the the nodes unscheduability = %t", false)}, metav1.ConditionUnknown, metav1.ConditionUnknown, node, ctrl.client, ctrl.featureGatesAccessor)
+		//	ctrl.EmitUpgradeEvent(ctrl.stateControllerPod, ctrl.UpgradeAnnotations(mcfgv1.MachineConfigNodeUpdateComplete, node), corev1.EventTypeNormal, "CordoningNode", fmt.Sprintf("Cordining Node %s as part of update", node.Name))
+		if err != nil {
+			klog.Errorf("Error making MCN: %w", err)
+		}
 		ctrl.logNode(node, "cordoning")
 		// perform cordon
 		if err := ctrl.cordonOrUncordonNode(true, node, drainer); err != nil {
 			return fmt.Errorf("node %s: failed to cordon: %w", node.Name, err)
 		}
 		ctrl.ongoingDrains[node.Name] = time.Now()
+		err = upgrademonitor.GenerateAndApplyMachineConfigNodes(&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateExecuted, Reason: string(v1alpha1.MachineConfigNodeUpdateCordoned), Message: fmt.Sprintf("(Un)Cordoned Node as part of In progress upgrade phase")},
+			&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateCordoned, Reason: fmt.Sprintf("%s%s", string(v1alpha1.MachineConfigNodeUpdateExecuted), string(v1alpha1.MachineConfigNodeUpdateCordoned)), Message: fmt.Sprintf("(Un)Cordoned node. The node is reporting Unschedulable = %t", node.Spec.Unschedulable)},
+			metav1.ConditionUnknown,
+			metav1.ConditionTrue,
+			node,
+			ctrl.client,
+			ctrl.featureGatesAccessor,
+		)
+		if err != nil {
+			klog.Errorf("Error making MCN: %w", err)
+		}
 	}
 
 	// Attempt drain
 	ctrl.logNode(node, "initiating drain")
+	err := upgrademonitor.GenerateAndApplyMachineConfigNodes(
+		&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateExecuted, Reason: string(v1alpha1.MachineConfigNodeUpdateDrained), Message: fmt.Sprintf("Draining Node as part of In progress upgrade phase")},
+		&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateDrained, Reason: fmt.Sprintf("%s%s", string(v1alpha1.MachineConfigNodeUpdateExecuted), string(v1alpha1.MachineConfigNodeUpdateDrained)), Message: fmt.Sprintf("Draining node. The drain will not be complete until desired drainer %s matches current drainer %s", node.Annotations[constants.DesiredDrainerAnnotationKey], node.Annotations[constants.LastAppliedDrainerAnnotationKey])},
+		metav1.ConditionUnknown,
+		metav1.ConditionUnknown,
+		node,
+		ctrl.client,
+		ctrl.featureGatesAccessor,
+	)
+	if err != nil {
+		klog.Errorf("Error making MCN: %w", err)
```

Comment:
Maybe we can reword error from making to either generating or creating

Reply:
Will do, Jerry suggested the same